### ISSUE TITLE
Multi-machine setup not working

### ISSUE THREAD

**Issue by mckenziec** [cockpit-project/cockpit#13060]
Cockpit version: 194
OS: AtomicOS (CentOS 7.1)

Hi. I'm trying to get the multi-server management working on CentOS but when I manually add a machines.d json and try to connect to the remote server, it fails with "Cockpit is not installed", "A compatible version of Cockpit is not installed on 192.168.177.171."

My setup consists of 2 Atomic/Centos servers/vms, with the same cockpit bridge and a cockpit-ws docker container which exposes port 9090. I added the following json to both under /etc/cockpit/machines.d/deployment.json:
```
{
    "atomic1.local": {
        "address": "192.168.177.171",
        "visible": true,
        "color": "blue",
        "user": "admin"
    },
    "atomic2.local": {
        "address": "192.168.177.172",
        "visible": true,
        "color": "green",
        "user": "admin"
    }
}
```

On Atomic there's a known issue about the dashboard plugin not being installable/available (or so my coworker tells me), so I simply threw the same machines.d json file on both.

When I refresh the cockpit browser view, there's a drop down icon when to the default server item in the menu, it shows both atomic1 and atomic2 servers. when I click on them I get "Couldn't connect to the machine, cannot connect to an unknown machine". I click Troublshoot and see  "Cockpit is not installed", "A compatible version of Cockpit is not installed on 192.168.177.171."

Does multi-machine require access to ssh only? Or also a different cockpit-ws port? Is this related to me running cockpit-ws inside docker and not exposing a port?

When I looked through the github project source, there's the suggestion that the cockpit bridge versions aren't the same, but they are both the same 195.1 as shipped with Atomic/Centos.

I disabled firewalld as well on both. I can ping and ssh between each other. I even accepted each other server's ssh keys as known by sshing from each other with the same user.

The admin user on both servers has the same password as well.

Any tips? Thanks.

**Comment by mckenziec** [comment]
Both servers have "PasswordAuthentication yes" in /etc/ssh/sshd_config (thanks to Atomic pre-setting it). 

Also, not sure if this helps:
```
$ cockpit-bridge --packages
base1                                                         /usr/share/cockpit/base1
docker               Containers                               /usr/share/cockpit/docker
domain                                                        /usr/share/cockpit/realmd
kdump                Kernel Dump                              /usr/share/cockpit/kdump
network              Networking                               /usr/share/cockpit/networkmanager
performance                                                   /usr/share/cockpit/tuned
selinux              SELinux                                  /usr/share/cockpit/selinux
shell                                                         /usr/share/cockpit/shell
sosreport            Diagnostic Reports                       /usr/share/cockpit/sosreport
system               Services, System, Logs, Terminal         /usr/share/cockpit/systemd
unity-deployment                                              /var/unity/cockpit/unity-deployment
unity-features       Features                                 /var/unity/cockpit/unity-features
unity-sample         Unity Sample                             /var/unity/cockpit/unity-sample
unity-syslog         Unity Syslog                             /var/unity/cockpit/unity-syslog
updates              Software Updates                         /usr/share/cockpit/cockpit-ostree
users                Accounts                                 /usr/share/cockpit/users
```

**Comment by mckenziec** [comment]
Is there anything I can dig into, like logs or attempts?

**Comment by mckenziec** [comment]
I'm shooting in the dark here but I generated some unprotected ssh keys for my admin user and copied them to both servers under the same user home directory. No luck.

Is the dashboard rpm/plugin required to do multi-server? I didn't think it was needed. I have a planned automation step that'll populate a machines.d json file. Atomic has a problem with the dashboard plugin I believe, but if its necessary, please let me know and I'll look into it further.

Thanks.

**Comment by mckenziec** [comment]
Ok I ran tcpdump on one of the servers and tried adding that server to the other in cockpit's dashboard. There's a clear back and forth to each other's ssh/22 port. In the below you can see .171 reach out to .172, then later .172 talk back to .171, and so on. Would that happen if authentication actually failed? Normally when a client connects to sshd you're not going to see the server try to talk to the client's port 22. So this is clearly a cockpit ssh based protocol you designed. 

Can I get some tips on what logs to look at?

Here's a bit of the dump. I didn't code block it so I could highlight the back and forth:
11:04:57.458599 IP **192.168.177.171.48894 > 192.168.177.172.22**: Flags [S], seq 1502211406, win 29200, options [mss 1460,sackOK,TS val 85382417 ecr 0,nop,wscale 7], length 0
11:04:57.458714 IP 192.168.177.172.22 > 192.168.177.171.48894: Flags [S.], seq 3100011125, ack 1502211407, win 28960, options [mss 1460,sackOK,TS val 201542 ecr 85382417,nop,wscale 7], length 0
11:04:57.459299 IP 192.168.177.171.48894 > 192.168.177.172.22: Flags [.], ack 1, win 229, options [nop,nop,TS val 85382418 ecr 201542], length 0
11:04:57.468790 IP 192.168.177.172.22 > 192.168.177.171.48894: Flags [P.], seq 1:22, ack 1, win 227, options [nop,nop,TS val 201551 ecr 85382418], length 21
11:04:57.469448 IP **192.168.177.171.48894 > 192.168.177.172.22**: Flags [.], ack 22, win 229, options [nop,nop,TS val 85382428 ecr 201551], length 0
11:04:57.469476 IP 192.168.177.171.48894 > 192.168.177.172.22: Flags [P.], seq 1:22, ack 22, win 229, options [nop,nop,TS val 85382428 ecr 201551], length 21
11:04:57.469492 IP 192.168.177.172.22 > 192.168.177.171.48894: Flags [.], ack 22, win 227, options [nop,nop,TS val 201553 ecr 85382428], length 0
11:04:57.471259 IP 192.168.177.172.22 > 192.168.177.171.48894: Flags [P.], seq 22:1302, ack 22, win 227, options [nop,nop,TS val 201554 ecr 85382428], length 1280
11:04:57.471844 IP 192.168.177.171.48894 > 192.168.177.172.22: Flags [P.], seq 22:470, ack 1302, win 251, options [nop,nop,TS val 85382430 ecr 201554], length 448
...

**Comment by mckenziec** [comment]
Alright! Sorry about the spam! PasswordAuthentication was not actually enabled. Hear me out. 

First the way I tested password auth was to use sshpass:
```
sshpass -p password ssh -o PreferredAuthentications=password -o PubkeyAuthentication=no -o StrictHostKeyChecking=no admin@192.168.177.172
```
It would fail. So I started digging into the sshd config running on Atomic. Well cockpit's service list actually helped me figure out that sshd was running the systemd unit file /usr/lib/systemd/system/sshd.service. The sshd unit file isn't included under /etc/systemd/system path.

The unit file pulls it's config variables from /etc/sysconfig/sshd:
EnvironmentFile=/etc/sysconfig/sshd

Which itself is very barebones. Once I threw "PasswordAuthentication yes" in there, machine2machine all worked fine.

Again, sorry for the spam issue, I know you guys have been working on a release this week. Maybe this will help someone else in the future.
